parsers/CA_ON.py

```diff
@@ -64,331 +55,265 @@
     "PQ.Q4C": "CA-ON->CA-QC",
     "PQ.X2Y": "CA-ON->CA-QC",
 }
-
-PRODUCTION_URL = "http://reports.ieso.ca/public/GenOutputCapability/PUB_GenOutputCapability_{YYYYMMDD}.xml"
+EXCHANGE_URL = "http://reports.ieso.ca/public/IntertieScheduleFlow/PUB_IntertieScheduleFlow_{YYYYMMDD}.xml"
+# Map IESO's production modes to ours.
+MODES = {
+    "BIOFUEL": "biomass",
+    "GAS": "gas",
+    "HYDRO": "hydro",
+    "NUCLEAR": "nuclear",
+    "SOLAR": "solar",
+    "WIND": "wind",
+}
+NAMESPACE = "{http://www.theIMO.com/schema}"
 PRICE_URL = (
     "http://reports.ieso.ca/public/DispUnconsHOEP/PUB_DispUnconsHOEP_{YYYYMMDD}.xml"
 )
-EXCHANGES_URL = "http://reports.ieso.ca/public/IntertieScheduleFlow/PUB_IntertieScheduleFlow_{YYYYMMDD}.xml"
-
-XML_NS_TEXT = "{http://www.theIMO.com/schema}"
+PRODUCTION_URL = "http://reports.ieso.ca/public/GenOutputCapability/PUB_GenOutputCapability_{YYYYMMDD}.xml"
+SOURCE = "ieso.ca"
+# IESO says "Eastern Standard Time is used year round." This means daylight
+# savings is not used (that is called "Eastern Daylight Time"), and we need to
+# use UTC-5 rather than 'Canada/Eastern'.
+TIMEZONE = datetime.timezone(datetime.timedelta(hours=-5), name="UTC-5")
+ZONE_KEY = ZoneKey("CA-ON")
 
 
-def _fetch_ieso_xml(
-    target_datetime: datetime | None,
-    session: Session | None,
+def _fetch_xml(
     logger: Logger,
-    url_template,
-):
-    dt = (
-        arrow.get(target_datetime)
-        .to(tz_obj)
-        .replace(hour=0, minute=0, second=0, microsecond=0)
+    session: Session | None,
+    target_datetime: datetime.datetime | None,
+    template: str,
```

Comment:
I would rename this to make it more clear, perhaps even split out the URL formatting from the fetch function.